version: '3.8'

services:
  app:
    build: .
    ports:
      - "8080:8080"
    environment:
      - PORT=8080
      - CELERY_BROKER_URL=redis://redis:6379/0
      - CELERY_RESULT_BACKEND=redis://redis:6379/0
      - PYTHONPATH=/app/src
    volumes:
      - ./output:/app/output
      - ./cache:/app/cache
      - ./torch:/app/torch
      - ./config:/app/config
    command: gunicorn --bind 0.0.0.0:8080 --workers 2 --worker-class gevent --timeout 300 --limit-request-line 8190 --limit-request-field_size 8190 main:app
    depends_on:
      - redis
    restart: unless-stopped
    mem_limit: 750m  # Увеличено, так как приложение падало по OOM
    shm_size: '1gb'   # Добавлен общий объем памяти для контейнера

  redis:
    image: redis:latest
    restart: unless-stopped
    ports:
      - "6379:6379"

  celery:
    build: .
    command: celery -A celery_app worker --loglevel=info
    environment:
      - CELERY_BROKER_URL=redis://redis:6379/0
      - CELERY_RESULT_BACKEND=redis://redis:6379/0
    volumes:
      - ./output:/app/output
      - ./cache:/app/cache
      - ./torch:/app/torch
      - ./config:/app/config
    depends_on:
      - redis
    restart: unless-stopped
    mem_limit: 750m
    shm_size: '1gb'   # Аналогично для Celery
